assets/src/edit-story/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,171 @@
+} from '../../../../../../app/media/utils';
+import { withProtocol } from '../../../../../../utils/url';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const { allowedFileTypes } = useConfig();
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+
+  const [errorMsg, setErrorMsg] = useState(false);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
+    description
+  );
+  const [link, setLink] = useState('');
+
+  const getFileInfo = useCallback(
+    (value = link) => {
+      const ext = value.split(/[#?]/)[0].split('.').pop().trim();
+      if (!allowedFileTypes.includes(ext)) {
+        setErrorMsg(error);
+      } else {
+        setErrorMsg(null);
+      }
+      return {
+        ext,
+        type: ['m4v', 'mp4'].includes(ext) ? 'video' : 'image',
+      };
+    },
+    [link, allowedFileTypes, error]
+  );
+
+  const onInsert = useCallback(async () => {
+    if (errorMsg?.length) {
+      return;
+    }
+    try {
+      const { type } = getFileInfo();
+      const isVideo = type === 'video';
+      const getMediaDimensions = isVideo
+        ? getVideoDimensions
+        : getImageDimensions;
+      const { width, height } = await getMediaDimensions(link);
+
+      // Create a poster file in case of videos.
+      let posterData;
+      if (isVideo) {
+        const originalFileName = getFileNameFromUrl(link);
+        const fileName = getPosterName(originalFileName);
+        const posterFile = await getFirstFrameOfVideo(link);
+        posterData = await uploadVideoPoster(0, fileName, posterFile);
+      }
+      insertElement(type, {
+        resource: createResource({
+          alt: link.substring(link.lastIndexOf('/') + 1),
+          width,
+          height,
+          src: link,
+          local: false,
+          poster: isVideo ? posterData.poster : null,
+          posterId: isVideo ? posterData.posterId : null,
+        }),
+      });
+      setErrorMsg(null);
+      setLink('');
+      onClose();
+    } catch (e) {
+      setErrorMsg(
+        __(
+          'Media failed to load. Please ensure the link is valid and the site allows linking from external sites',
+          'web-stories'
+        )
+      );
+    }
+  }, [insertElement, link, errorMsg, getFileInfo, onClose, uploadVideoPoster]);
```

Comment:
(Just a general note)

Post-MVP we might need to refactor this to work more async, similar to how the media upload queue works. This would also help with CORS.

Example:

1. Insert URL & confirm
2.`onInsert` directly adds a placeholder resource to the canvas
3. We try to fetch the image/video and get more information
4. Update placeholder with new information (media type, dimensions)
5. If information gathering fails (e.g. because of CORS), we can update the placeholder to say "Preview unavailable" or something

Reply:
Added a note about this to the follow up issue: #8653